pkg/agent/agent.go

```diff
@@ -404,6 +412,14 @@ func startStreamMode(agent agentData) error {
 		agent.proxyResourceHandler,
 	}
 
+	// Register managed application and access handler for traceability agent
+	// For discovery agent, the handlers gets registered while setting up provisioner
+	agentType := agent.cfg.GetAgentType()
+	if agentType == config.TraceabilityAgent {
+		handlers = append(handlers, handler.NewAccessRequestHandler(nil, agent.cacheManager, agent.apicClient, agentType))
+		handlers = append(handlers, handler.NewManagedApplicationHandler(nil, agent.cacheManager, agent.apicClient, agentType))
```

Comment:
This is kind of what I was thinking of when I mentioned a specific handler for trace mode for the access request handler. Break the trace part of both resources into their own handlers and only register it in trace, and only register the discovery part in discovery.

Reply:
Done